x/bank/keeper/keeper.go

```diff
@@ -53,6 +53,29 @@ type BaseKeeper struct {
 	paramSpace paramtypes.Subspace
 }
 
+func (k BaseKeeper) GetPaginatedTotalSupply(ctx sdk.Context, pagination *query.PageRequest) (sdk.Coins, *query.PageResponse, error) {
+	store := ctx.KVStore(k.storeKey)
+	supplyStore := prefix.NewStore(store, types.SupplyKey)
+
+	supply := sdk.NewCoins()
+
+	pageRes, err := query.Paginate(supplyStore, pagination, func(_, value []byte) error {
+		var result sdk.Coin
+		err := k.cdc.UnmarshalBinaryBare(value, &result)
+		if err != nil {
+			return err
+		}
+		supply = append(supply, result)
+		return nil
+	})
+
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return supply, pageRes, nil
+}
+
 func (k BaseKeeper) GetTotalSupply(ctx sdk.Context) sdk.Coins {
```

Comment:
How about we go ahead and delete? Wdyt @sahith-narahari ?

Reply:
agree, I had the same plan https://github.com/cosmos/cosmos-sdk/pull/8798#discussion_r599546953. I'll make the changes